### ISSUE TITLE
Failed to load tplg on GLK

### ISSUE THREAD

**Issue by keqiaozhang** [thesofproject/linux#322]
Topology load failed on GLK.
This regression may caused by firmware PR: thesofproject/sof/pull/634 and some related changes in driver side are missed.

Dmesg error:
```
[    4.900960] sof-audio sof-audio: tplg: config HDA4 fmt 0x0
[    4.900986] sof-audio sof-audio: ipc tx: 0x80010000
[    4.901086] sof-audio sof-audio: error: ipc error for 0x80010000 size 0xc
[    4.901095] sof-audio sof-audio: error: failed to set DAI config for HDA4
[    4.901097] sof-audio sof-audio: ASoC: physical link loading failed
[    4.901100] sof-audio sof-audio: error: tplg component load failed -5
[    4.901110] sof-audio sof-audio: error: failed to load DSP topology -22
[    4.901113] sof-audio sof-audio: ASoC: failed to probe component -22
[    4.901143] sof-audio sof-audio: ipc rx: 0x90020000
[    4.901152] bxt_da7219_max98357a glk_da7219_max98357a: ASoC: failed to instantiate card -22
[    4.901163] bxt_da7219_max98357a glk_da7219_max98357a: Topology forgot to remove link Speakers?
[    4.901164] sof-audio sof-audio: ipc rx done: 0x90020000
[    4.901167] bxt_da7219_max98357a glk_da7219_max98357a: Topology forgot to remove link Headset?
[    4.901170] bxt_da7219_max98357a glk_da7219_max98357a: Topology forgot to remove link DMIC01?
[    4.901172] bxt_da7219_max98357a glk_da7219_max98357a: Topology forgot to remove link HDMI1?
[    4.901174] bxt_da7219_max98357a glk_da7219_max98357a: Topology forgot to remove link HDMI2?
[    4.901177] bxt_da7219_max98357a glk_da7219_max98357a: Topology forgot to remove link HDMI3?
[    4.906633] bxt_da7219_max98357a: probe of glk_da7219_max98357a failed with error -22
```

Trace log:
sudo sof-logger -l /root/sof-apl.ldc
```
CORE  LEVEL      COMP_ID                TIMESTAMP            DELTA                FILE_NAME    CONTENT
    0      1          DMA            231250.625000    231250.625000            hda-dma.c:448    hda-dmac: 7 no free channel 0
    0      1          DAI            231256.614583         5.989583                dai.c:734    dai_config() error: dma_channel_get() failed
    0      1          IPC            231261.718750         5.104167                ipc.c:361    eCD
```
sof-logger -t:
```
    0      2          DAI            230993.697917         3.437500                dai.c:708    dai_config(), dev->frame_bytes = 8
    0      2          DMA            230997.187500         3.489583             dw-dma.c:332    dw-dma: 0 channel 0 request
    0      2          IPC            231096.406250        99.218750            apl-ipc.c:114    ipc: msg rx -> 0x10000
    0      2          IPC            231102.239583         5.833333            handler.c:492    ipc: dai 3,3 -> config
    0      2          DAI            231106.406250         4.166667                dai.c:86     dai-get 0xbe04c3d4 sref 2
    0      2          DAI            231110.208333         3.802083                dai.c:110    dai-put 0xbe04c3d4 sref 1
    0      2          DMA            231121.666667        11.458333            hda-dma.c:432    hda-dmac: 7 channel 0 -> get
    0      2          IPC            231221.093750        99.427086            apl-ipc.c:114    ipc: msg rx -> 0x10000
    0      2          IPC            231226.562500         5.468750            handler.c:492    ipc: dai 3,4 -> config
    0      2          DAI            231230.572917         4.010417                dai.c:86     dai-get 0xbe04c420 sref 2
    0      2          DAI            231234.322917         3.750000                dai.c:110    dai-put 0xbe04c420 sref 1
    0      2          DMA            231246.979167        12.656250            hda-dma.c:432    hda-dmac: 7 channel 0 -> get
    0      1          DMA            231250.625000         3.645833            hda-dma.c:448    hda-dmac: 7 no free channel 0
```

Tested with:
Tested with:
kernel-cros-sof-v4.14-rebase: f8f388ef
sof-master: 31cde1
tplg-master: ad3f3e4
tplg:sof-glk-da7219.tplg
[dmesg.log](https://github.com/thesofproject/linux/files/2623186/dmesg.log)
[logger.log](https://github.com/thesofproject/linux/files/2623187/logger.log)
[logger-t.log](https://github.com/thesofproject/linux/files/2623188/logger-t.log)

**Comment by mengdonglin** [comment]
@keqiaozhang @RanderWang thesofproject/sof/pull/635 only revise WHL platform m4, why this can affect GLK and APL?

**Comment by mengdonglin** [comment]
Also reproduced on APL UP2 platform: https://github.com/thesofproject/sof/issues/644

**Comment by mengdonglin** [comment]
@RanderWang  @libinyang  I suspect thesofproject/sof/pull/634 or thesofproject/sof/pull/633 introduced the regression. 

@keqiaozhang @markyang Could you help to bisect?

**Comment by keqiaozhang** [comment]
@mengdonglin @RanderWang 
sorry, I pasted the wrong PR ID, the PR should be thesofproject/sof/pull/634

**Comment by mengdonglin** [comment]
@RanderWang  It seems https://github.com/thesofproject/linux/pull/291 is needed in kernel side.

**Comment by mengdonglin** [comment]
@keqiaozhang @markyang @RanderWang Could help to verify topology can be loaded now on GLK and APL (issue thesofproject/sof#644)?

**Comment by RanderWang** [comment]
@mengdong, sorry, bard borrowed my APL

**Comment by keqiaozhang** [comment]
@ranj063 
Please help to cherry-pick #291  to cros-sof-v4.14-rebase.
Verified that this PR can fix the topology loading issue on APL

**Comment by mengdonglin** [comment]
@keqiaozhang thank you for verifying APL! @markyang @RanderWang APL is okay now.

**Comment by markyang** [comment]
**Summary:**
  This issue cannot be reproduced on UP2 with PCM512x.

**Test env:**
sof master: e94a34fc
kernel sof-dev: de1f57ac
tplg: sof-apl-pcm512x.tplg

**Comment by ranj063** [comment]
@keqiaozhang cros branch has been updated. Please re-check this issue.

**Comment by plbossart** [comment]
no news -> closing.

**Comment by mengdonglin** [comment]
@keqiaozhang Could you help to check if this issue is reproducible on GLK?

**Comment by keqiaozhang** [comment]
Verified that this issue is fixed on cros-sof-v4.14-rebase/40fae8a8. Close this bug.

**Comment by markyang** [comment]
@stevyan
